js/views/modals/purchase/Purchase.js

```diff
@@ -128,28 +122,37 @@ export default class extends BaseModal {
     }
   }
 
-  changeModerator(data) {
-    if (data.selected) {
-      this.order.set('moderator', data.guid);
-    } else if (data.guid === this.order.get('moderator')) {
-      // the current moderator was deselected
-      this.order.set('moderator', '');
-    }
+  changeQuantityInput(e) {
+    this.order.get('items').at(0).set('quantity', $(e.target).val());
   }
 
-  changeQuantityInput(e) {
-    console.log(e);
-    // use a data attribute of the listingHash to associate this with the right item
+  clickPayBtn() {
+    this.$confirmPay.removeClass('hide');
   }
 
-  changeQuantity(quantity, listingHash) {
-    // change the quantity of the item
-    console.log(quantity);
-    console.log(listingHash);
+  clickConfirmBtn() {
+    // confirm the purchase
+    this.purchaseListing();
   }
 
-  clickPayBtn() {
-    console.log(this.order.attributes);
+  closeConfirmPay() {
+    this.$confirmPay.addClass('hide');
+  }
+
+  purchaseListing() {
+    // set the moderator
+    this.order.set('moderator', this.moderators.selectedIDs[0]);
+
+    $.post({
```

Comment:
This section is to test whether the order model data can be used to call the purchase api.